Resolved Issue #20 Not sending to Multiple Number

TheKoola opened the pull request:
Changes to `_sent_txt` function to cull individual "to" phone numbers separated by commas and loop over sending text message to each one.

TheKoola committed:
Resolved Issue #20 Not sending to Multiple Number

tedder commented on octoprint_smsnotifier/__init__.py:
```diff
@@ -101,37 +101,34 @@ def _send_txt(self, payload, snapshot=False):
+            }
+            message = self._settings.get(["message_format", "body"]).format(**tags)
+
+            client = TwilioRestClient(self._settings.get(['account_sid']), self._settings.get(['auth_token']))
```
you should be able to grab the Client outside of the loop.

TheKoola replied:
And set the tags and message outside the loop too.  Would be more efficient.

tedder replied:
heh, yeah, I saw that at the very end before seeing your comment :)

tedder commented on octoprint_smsnotifier/__init__.py:
```diff
@@ -101,37 +101,34 @@ def _send_txt(self, payload, snapshot=False):
+                    return False
+                else:
+                    # report notification was sent
+                    self._logger.info("Print notification sent to %s" % (self._settings.get(['recipient_number'])))
```
You've removed the "return true" entries on success. If it still returns true it's just a side effect (of self.logger). It appears that `on_event` uses that status (for instance, retrying text-only if sending an image fails).

If it were me, I'd `return true` from the "else" of the try blocks, then `return false` at the very end of the method. That's just a personal preference towards only returning success in known cases.

TheKoola replied:
But doesn't `return true` break out of the loop?  You only get the first (successful) phone number out of the list in that case...

tedder replied:
Oh! Duh, of course. I have an idea but I'm sure you do too.

TheKoola replied:
What do you prefer?  Close this pull request and start a new one?  Or close this one and have the contributors fix it?

tedder replied:
I like what you've done! It's just about how to track and return success or failure, so do it in here. By "I have an idea" I meant "if you get stuck on coding a return to this loop, let me know".

shadycuz replied:
@TheKoola @tedder 

I can't find anything in the [docs](https://docs.octoprint.org/en/master/modules/plugin.html?highlight=on_event#octoprint.plugin.types.EventHandlerPlugin) that says we need to return true or false when `on_event()` is called. We can keep it, but I just wouldn't worry about it.

tedder commented on octoprint_smsnotifier/__init__.py:
```diff
@@ -101,37 +101,34 @@ def _send_txt(self, payload, snapshot=False):
+                'elapsed_time': elapsed_time,
+                'printer_name': self._settings.get(["printer_name"])
+            }
+            message = self._settings.get(["message_format", "body"]).format(**tags)
```
actually, `tags` and `message` can be done outside the loop too.

TheKoola replied:
Agreed -- would be more efficient than doing it every loop iteration.

TheKoola committed:
Resolved Issue 20: Not sending to Multiple Number

TheKoola commented:
Updated with new commit:
* Moved `client`, `tags` and `messages` out of loop so only done once at beginning
* Added `return True` to end of function after loop since all messages have to be sent successfully to get there.  Any fail-to-send message will break loop and `return False`

tedder reviewed (approved):
nice!

shadycuz commented:
@TheKoola I'm unable to push to your fork =/ 

```shell
shadycuz-> git push --set-upstream koola fix/multiple_numbers 
Total 0 (delta 0), reused 0 (delta 0)
To github.com:TheKoola/OctoPrint-Twilio.git
 ! [remote rejected] fix/multiple_numbers -> fix/multiple_numbers (permission denied)
error: failed to push some refs to 'git@github.com:TheKoola/OctoPrint-Twilio.git'
```
I wanted to add some minor changes, but its not a big deal. I can pull your changes in after the python3 fix is confirmed. This fix will be the next thing out the door.

shadycuz commented:
@TheKoola Closing this for #24 only because I couldn't make some tweaks and bump the version on your branches =/ 

Don't worry your commits are still in place and you will get credit + that contributors tag 👍 

Thanks for fixing the bug and if you could review my PR, its a bit hard to see the diff correctly, might want to just look at the file.
